docs/howto/components.rst

```diff
@@ -0,0 +1,84 @@
+How to package and upload a snap with components
```

Comment:
I'm going to make a statement here that might open a can of worms: to me, this text is fine but reads more like a tutorial than a how-to. I'm actually fine with it because as long as we have the content *somewhere* we can always move it around later, just thought I'd bring it up!

Apparently these two doc types are mixed up often: https://diataxis.fr/tutorials-how-to/#tutorials-how-to

Reply:
When `snapd` allows for downloading and installing later this year, that might be a good time to make this more useful and reorganise it as a tutorial.